docs/DeveloperGuide.md

```diff
@@ -1,38 +1,335 @@
+
+1. **Configure the JDK**: Follow the guide [_[se-edu/guides] IDEA: Configuring the
+   JDK_](https://se-education.org/guides/tutorials/intellijJdk.html) to ensure Intellij is configured to use **JDK 11
+   **.
+2. **Import the project as a Gradle project**: Follow the guide [_[se-edu/guides] IDEA: Importing a Gradle
+   project_](https://se-education.org/guides/tutorials/intellijImportGradleProject.html) to import the project into
+   IDEA.<br>
+   :exclamation: Note: Importing a Gradle project is slightly different from importing a normal Java project.
+3. **Verify the setup**:
+    1. Run the `seedu.duke.Duke` and try a few commands.
+    2. Run the tests using `./gradlew check` to ensure they all pass.
+
+---
+
+### Before writing code
+
+1. **Configure the coding style**
+   If using IDEA, you can use the following steps to import the code style settings.
+    1. Go to `File → Settings → Editor → Code Style`
+    2. Click the Gear Icon next to the `Scheme` box and then click `Import Scheme → IntelliJ IDEA code style XML`.
+    3. Select the `DefaultCodeStyle.xml` file in the root of the project directory.
+2. **Set up CI**
+   This project comes with a GitHub Actions config files (in `.github/workflows` folder). When GitHub detects those
+   files, it will run the CI for your project automatically at each push to the `master` branch or to any PR. No set up
+   required.
+3. **Learn the design**
+   When you are ready to start coding, we recommend that you get some sense of the overall design by reading
+   about [MoneyMoover’s architecture](DeveloperGuide.md#architecture).
+
+When you are ready to start coding, we recommend that you get some sense of the overall design by reading
+about [MoneyMoover’s architecture](DeveloperGuide.md#architecture).
+
+## Design
+
+### Architecture
+
+![ArchitectureDiagram](images/ArchitectureDiagram.png)
+
+The diagram above provides a high-level overview of how the project is structured. The main components are:
+
+1. The `Main` class which initialises all the other components at startup, and connects them with each other
+2. The `UI` component which is responsible for all user input and output
+3. The `Parser` component which parses user input and creates the relevant Command objects
+4. The `Command` component which executes the logic
+5. The `Accounts` component which manages the user's accounts
+6. The `Forex` (Foreign Exchange) component which handles exchange-rate related logic
+7. The `Transactions` component which manages the user's transactions
+8. The `Storage` component which handles the saving and loading of data to disk
+
+### General Sequence
+
+The following is a high-level sequence of a single `create-account SGD` command, which demonstrates how the components
+interact with each other:
+
+![BasicSequence](images/BasicSequence.png)
+
+### UI component
+
+The API of this component is specified in the `Ui.java`.
+
+![UI Class Diagram](images/UiClassDiagram.png)
```

Comment:
Is the text on the enumeration part of the image too flushed to the left? Should there be some padding instead?

![image](https://user-images.githubusercontent.com/35862661/228457313-f82b30b2-c4ba-41d0-8e4d-449836205d28.png)